dds/DCPS/ReactorInterceptor.cpp

```diff
@@ -49,8 +49,25 @@ ReactorInterceptor::~ReactorInterceptor()
 
 bool ReactorInterceptor::should_execute_immediately()
 {
-  return ACE_OS::thr_equal(owner_, ACE_Thread::self()) ||
-    reactor_is_shut_down();
+  bool is_safe_to_execute = false;
```

Comment:
Go further.  Manually in-line all of the functions in `execute_or_enqueue` and then simplify the logic.